src/shogun/base/SGObject.cpp

```diff
@@ -711,12 +711,12 @@ bool CSGObject::equals(CSGObject* other, float64_t accuracy, bool tolerant)
 		SG_DEBUG("comparing parameter \"%s\" to other's \"%s\"\n",
 				this_param->m_name, other_param->m_name);
 
-		/* hard-wired exception for DynamicObjectArray parameter num_elements */
-		if (!strcmp("DynamicObjectArray", get_name()) &&
+		/* hard-wired exception for [Wrapped|Dynamic]ObjectArray parameter num_elements */
+		if ((strcmp("DynamicObjectArray", get_name()) == 0 || strcmp("WrappedObjectArray", get_name()) == 0) &&
```

Comment:
Ha, didn't know `DynamicObjectArray` had a child... And it was forgotten whenever this beauty here was written...